src/validation.cpp

```diff
@@ -4135,8 +4138,9 @@ bool CVerifyDB::VerifyDB(
         uiInterface.ShowProgress(_("Verifying blocks...").translated, percentageDone, false);
         if (pindex->nHeight <= chainstate.m_chain.Height()-nCheckDepth)
             break;
-        if (fPruneMode && !(pindex->nStatus & BLOCK_HAVE_DATA)) {
-            // If pruning, only go back as far as we have data.
+        if ((fPruneMode || is_snapshot_cs) && !(pindex->nStatus & BLOCK_HAVE_DATA)) {
```

Comment:
9b604c0207b526c008617cdca210f35db5e344db: Wouldn't it be easier to remove the prune and snapshot condition and simply check HAVE_DATA? Performance shouldn't decrease and it would be less code, which is nice.

Reply:
I guess your goal is to treat it as error when block data is missing, but pruning is off an no snapshot is loaded?

Reply:
Right - I think we want to fail here if we're lacking block data when not pruning or operating from snapshot.